.github/workflows/unit_test.yml

```diff
@@ -56,3 +56,30 @@ jobs:
           GOBIN: /home/runner/go/bin
     - name: Run 
       run: make test-verbose
+
+  integration_test:
+    runs-on: ubuntu-latest
+    needs: unit_test_without_bcc
+    steps:
+      - uses: actions/checkout@main
+      - uses: actions/setup-go@main
+        with:
+          go-version: 1.18
+      - name: install kind
+        run: go install sigs.k8s.io/kind@v0.12.0
+      - name: check version
+        run: kind version
+      - name: install kubectl
+        run: curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+      - name: check version
+        run: kubectl version || true
+      - name: build image
+        run: export CTR_CMD=docker && make
+      - name: infra up
+        run: ./scripts/kind.sh
+      - name: upload image to local registry
+        run: kind load docker-image quay.io/sustainable_computing_io/kepler:latest
+      - name: deploy test
+        run: kubectl create -f manifests/kubernetes/deployment.yaml 
```

Comment:
The `make cluster-sync` command deletes a running kepler instance, creates a new kepler image, pushes it to the repository and deploys kepler to the cluster with the newer image.

It will be great if you use this, which will reuse all the scripts already created for this...
Also, using the scripts we will make sure that everyone will be using the same flow...

Reply:
again, we can't reuse current script for now.
with reason
`make cluster-sync` is used for release with upstream and for this script just used for KIND as local k8s cluster for testing.
https://github.com/sustainable-computing-io/kepler/blob/main/Makefile#L105
we can't upload pr level image as a temp image to over right any publish registry without test it.

Reply:
That is why we need to carefully define the env variables:
`IMAGE_REPO="quay.io/sustainable_computing_io/kepler"` (already the default)
`IMAGE_TAG="$(git describe --tags)"`

But, you will need to create a new script to push the image, which will read the `CLUSTER_PROVIDER` and if it is `"kind"`, it will push the image to the kind cluster

Note that the `build-manifest` will already generate the yam `_output/manifests/kubernetes/generated/deployment.yaml` with the custom image...